### PR TITLE
Protects metrics route with basic auth

### PR DESCRIPTION
**Pull request by askldjd**
Uses the Rack Basic Auth middleware to protect the `/metrics` route that's scraped by Prometheus.

Testing:
`/metrics` now requires authentication.

To do an integrated test, we'll first merge https://github.com/department-of-veterans-affairs/appeals-deployment/pull/351. Then merge this, then verify that Prometheus still works.

### CODE REVIEW

**Comment by alex** [comment]
I'm not the right Alex, but I just wanted to share how proud and joyfoul I feel knowing that Caseflow now has metrics and all the other good things I see going on :-)

**Comment by askldjd** [comment]
Ah, crap. Sorry @alex. It has been a long day. Have a great holiday!

**Comment by alex** [comment]
🎉 ☃️ 🎉

File: config.ru [hunk 1]
```diff
@@ -5,7 +5,28 @@ require "rack"
 require "prometheus/client/rack/collector"
 require "prometheus/client/rack/exporter"
 
+# Perform basic auth on the Prometheus /metrics endpoint so that we do not
+# expose sensitive data in the open.
+class MetricsAuth < Rack::Auth::Basic
+  def call(env)
+    request = Rack::Request.new(env)
+    case request.path
+
```
Comment (shanear) [inline config.ru]: why the newline lol? makes the case statement hard to read.

Comment (askldjd) [inline config.ru]: Oh hmmmm.... No idea... Fat fingers

**Comment by amprokop** [comment]
looks like we'll have to add the username/password to our `scrape_config` in Prometheus, so Prometheus sets the `Authorization` header on every request to the Rails `/metrics` endpoint.

https://github.com/prometheus/docs/blob/master/content/docs/operating/configuration.md

Taking over for Alan on this.

**Comment by amprokop** [comment]
@shanear — can you take a look at this again?

File: config.ru [hunk 2]
```diff
@@ -5,17 +5,22 @@ require "rack"
...
 require "prometheus/client/rack/exporter"
 
-if Rails.env.development? || Rails.env.demo?
-  # use gzip for the '/metrics' route, since it can get big.
-  use Rack::Deflater,
-      if: -> (env, _status, _headers, _body) { env["PATH_INFO"] == "/metrics" }
+# require basic auth for the /metrics route
+use MetricsAuth, "metrics" do |username, password|
+  # if metrics password mistakenly isn't provided, disable the route
+  return false if ENV["METRICS_PASSWORD"].blank?
```
Comment (askldjd) [inline config.ru]: I don't think you can `return` from a block in ruby. When I run this, I get a LocalJumpError. Maybe we should check the existence of the env before using this middleware.

Comment (amprokop) [inline config.ru]: Agh. Thanks for the thorough testing. I guess I always tested where "METRICS_PASSWORD" was defined. Willfix.

Comment (amprokop) [inline config.ru]: image
reproduced, thanks

**Comment by amprokop** [comment]
Fixed, and tested that when no password is defined in the environment, access is not permitted. Is the refactor I did ok with you, @askldjd?

**Comment by askldjd** [comment]
:+1: Looks great. Sorry for taking so long to review this.

**Pull request merged**